src/plugins/intel_cpu/src/shape_inference/static_shape.hpp

```diff
@@ -41,6 +41,8 @@ constexpr bool is_static_shape_adapter() {
     return std::is_same_v<U, StaticShapeRef> || std::is_same_v<U, StaticShape>;
 }
 
+// NOLINTBEGIN(google-explicit-constructor)
```

Comment:
Please add a reason

Reply:
Here, also, static shape adapter allows implicit conversion for the shapes that leads to less code. Otherwise, the code is unnecessarily verbose.